### ISSUE TITLE
Promote External Secret Stores to Beta

### ISSUE THREAD

**Issue by turkenh** [crossplane/crossplane#4164]
### What problem are you facing?

We released initial support for [External Secret Stores](https://github.com/crossplane/crossplane/blob/master/design/design-doc-external-secret-stores.md) in the [v1.7.0 milestone](https://github.com/crossplane/crossplane/releases/tag/v1.7.0). As a follow-up, we have made them [pluggable](https://github.com/crossplane/crossplane/blob/master/design/one-pager-pluggable-secret-stores.md) to enable out-of-tree development and extension with [v1.12.0](https://github.com/crossplane/crossplane/releases/tag/v1.12.0) and released the first out-of-tree external secret store plugin under https://github.com/crossplane-contrib/ess-plugin-vault.  

We want to continue to invest in this feature to evolve it to a Beta level of maturity.

### How could Crossplane help solve your problem?

This issue will serve as a tracking epic of the known issues we want to address before moving this feature to Beta.

```[tasklist]
### ESS Beta checklist
- [ ] https://github.com/crossplane/crossplane/issues/2985
- [ ] https://github.com/crossplane/crossplane/issues/4018
- [ ] https://github.com/crossplane/crossplane/issues/2998
- [ ] https://github.com/crossplane/crossplane/issues/2774
- [ ] At least 1 more ESS plugin has been developed and published in the ecosystem.
- [ ] https://github.com/crossplane/crossplane/issues/4171
- [ ] https://github.com/crossplane/crossplane/issues/4185
- [ ] https://github.com/crossplane/crossplane/issues/3361
- [ ] https://github.com/crossplane/crossplane/issues/4277
- [ ] https://github.com/crossplane/crossplane/issues/4725
```

**Comment by turkenh** [comment]
Removing deprecated Vault API from ESS API https://github.com/crossplane/crossplane-runtime/pull/513

**Comment by mmassenzio** [comment]
We've been using the `publishConnectionDetailsTo` feature to our corp Vault instance for several months now and it has proved pretty solid and really reliable, so my vote goes for promoting it
:100:

**Comment by EricDotz** [comment]
Hi, do know about a new version of the **ess-plugin-vault** (https://github.com/crossplane-contrib/ess-plugin-vault) with **Kubernetes AuthMethod**?

I saw this functionality in the code and would like to use it.

The code is already on the main branch.

**Comment by ezgidemirel** [comment]
Hi @EricDotz, just released `ess-plugin-vault:0.2.0`.
